pkg/operator/controller/ingress/deployment.go

```diff
@@ -498,6 +514,11 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: WildcardRouteAdmissionPolicy, Value: "false"})
 	}
 
+	if HTTP2IsDisabledByAnnotation(ci.Annotations) || HTTP2IsDisabledByAnnotation(ingressConfig.Annotations) {
```

Comment:
What if the cluster ingress config has `ingress.operator.openshift.io/unsupported-disable-http2=true` and the ingresscontroller has `ingress.operator.openshift.io/unsupported-disable-http2=false`?  Seems like the ingresscontroller's annotation should override the ingress config's.  Granted, this is an unsupported annotation, so I wouldn't worry too much about this if it makes the logic hairy.

Reply:
> Seems like the ingresscontroller's annotation should override the ingress config's.

Interesting. I interpreted the ingress config to have the higher priority. Rather than annotating all the ingresscontrollers you could just add it to the ingress config to disable it everywhere.

Reply:
My reasoning is that if the user explicitly added an annotation to both the ingress config as well as to an individual ingresscontroller, then the user is expressing a general preference with the former annotation and an overriding preference for the specific ingresscontroller with the latter annotation.